### ISSUE TITLE
Unable to connect Mongo DB Atlas

### ISSUE THREAD

**Issue by shabbirlathsaheb-gep** [spaceuptech/space-cloud#1365]
Tried adding the mongo db using the following format. it gives below error:
mongodb+srv://<yourusername>:<passowrd>@<atlas server>/<database name>?retryWrites=true&w=majority

Error adding database
Unable to set crud config

Please help.

**Comment by sharadregoti** [comment]
Hey @shabbirlathsaheb-gep 
1) which version of space cloud are you running?
2) Can you provide the logs of gateway container. 
if you are using docker, the command is `docker logs "containerName"`

**Comment by shabbirlathsaheb-gep** [comment]
> Hey @shabbirlathsaheb-gep
> 
> 1. which version of space cloud are you running?
> 2. Can you provide the logs of gateway container.
>    if you are using docker, the command is `docker logs "containerName"`

Hi. I am able to connect Mongodb now. thanks for prompt response.

**Comment by alichry** [comment]
Hello,

I'm currently facing the same issue. If I try to add a MongoDB, the error "Unable to set crud config" appears.

Here's an extract of the gateway log:

```json
[
  {
    "L": "ERROR",
    "T": "2020-10-25T11:56:49.657Z",
    "C": "helpers@v0.1.2/http.go:22",
    "M": "Unable to set crud config",
    "error": null,
    "requestId": "1jMvvb34conRiF8E9nHpI77Rjm7",
    "S": "github.com/spaceuptech/helpers.(*logger).LogError\n\t/go/pkg/mod/github.com/spaceuptech/helpers@v0.1.2/logger.go:83\ngithub.com/spaceuptech/helpers.(*response).SendErrorResponse\n\t/go/pkg/mod/github.com/spaceuptech/helpers@v0.1.2/http.go:22\ngithub.com/spaceuptech/space-cloud/gateway/server/handlers.HandleSetDatabaseConfig.func1\n\t/build/server/handlers/config_crud.go:164\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2007\ngithub.com/gorilla/mux.(*Router).ServeHTTP\n\t/go/pkg/mod/github.com/gorilla/mux@v1.7.3/mux.go:212\ngithub.com/spaceuptech/space-cloud/gateway/server.(*Server).restrictDomainMiddleware.func1\n\t/build/server/http.go:30\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2007\ngithub.com/spaceuptech/space-cloud/gateway/server.loggerMiddleWare.func1\n\t/build/server/middleware.go:29\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2007\ngithub.com/rs/cors.(*Cors).Handler.func1\n\t/go/pkg/mod/github.com/rs/cors@v1.7.0/cors.go:219\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2007\ngithub.com/mholt/certmagic.(*Config).HTTPChallengeHandler.func1\n\t/go/pkg/mod/github.com/mholt/certmagic@v0.9.1/httphandler.go:37\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2007\nnet/http.serverHandler.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2802\nnet/http.(*conn).serve\n\t/usr/local/go/src/net/http/server.go:1890"
  },
  {
    "L": "INFO",
    "T": "2020-10-25T11:56:49.657Z",
    "C": "helpers@v0.1.2/http.go:32",
    "M": "Response",
    "requestId": "1jMvvb34conRiF8E9nHpI77Rjm7",
    "fields": {
      "statusCode": 500
    }
  }
]
```

**Comment by sharadregoti** [comment]
@alichry Can you share the format of your MongoDB connection string?

**Comment by alichry** [comment]
`mongodb://mylocalhost.local:27017`

I've replaced the hostname with the IP address and it worked. The problem is due to docker using the same hostname as the host machine in containers. The specified hostname resolves to 127.0.0.1 on the gateway container.

Thank you for responding quickly :)
